facebookresearch/ParlAI
# Script for Chatbot using Slack

**Patch description**
Adds a new file  `parlai/scripts/interactive_slack.py` that is based on interactive_web but instead communicates with a Slack app bot to observe and act. Just pass the script a bot token and it will do the rest

**Testing steps**
Create a Slack App with legacy bot tokens that can read messages from channels in real time. See here for instructions on creating the app: https://ritikjain1272.medium.com/how-to-make-a-slack-bot-in-python-using-slacks-rtm-api-335b393563cd
Set the bot token env as `SLACK_BOT_TOKEN` or pass `--token` to the script.
Run the following
`parlai interactive_slack --token xoxb-... --task blended_skill_talk:all -mf zoo:blenderbot2/blenderbot2_400M/model --search-server http://localhost:5000/search`
Open a new channel in the slack workspace that you installed the bot in and once it's invited to the conversation (through `@` mention) you can just start talking to it in the channel

**Other information**
I know I probably should have done this in the chat_service providers alongside Telegram/etc but this just seemed easier to me. Depends on the `slackclient` python package
Tested working on BlenderBot 2.0. Screenshots below
<img width="473" alt="Screen Shot 2021-07-20 at 4 36 46 PM" src="https://user-images.githubusercontent.com/35289/126393224-d1414d19-a1d6-48d4-b71b-4bc92ecca867.png">


**Search Server**
For BlenderBot 2 i have implemented a domain specific search server in Flask and Elasticsearch to help the slackbot return relevant information. The project can be found below and includes instructions on loading all of the Slack help articles to give the bot some domain specific info about the product. Can be applied to other domain specific indexes (eg company intranet)

https://github.com/spacetelescope/learning/tree/main/chatbot
